### PR TITLE
Update to rocksdb 0.14 and set max wal size

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Rocksdb has too-high wal size causing long restart times.

#### Summary of Changes

Update to 0.14 with wal-size exposed and set it to 4gb.

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
@sakridge - I added the v1.1 backport label because I want to see how this behaves on our TdS node.  I wonder if we'll see any performance impact by reducing the WAL size though

**Comment by sakridge** [comment]
> 
> 
> @sakridge - I added the v1.1 backport label because I want to see how this behaves on our TdS node. I wonder if we'll see any performance impact by reducing the WAL size though

Ok, I'm also running some perf tests on master with the shred-fs logic reverted.

File: ledger/src/blockstore_db.rs [hunk 1]
```diff
@@ -844,5 +843,9 @@ fn get_db_options() -> Options {
     options.create_missing_column_families(true);
     // A good value for this is the number of cores on the machine
     options.increase_parallelism(num_cpus::get() as i32);
+
+    // Set max total wal size to 4G.
+    options.set_max_total_wal_size(4 * 1024 * 1024);
```
Comment (ryoqun) [inline ledger/src/blockstore_db.rs:848]: @sakridge `4 * 1024 * 1024` (= argument) doesn't match to 4G (= comment). Well, is `set_max_total_wal_size` is expected to take the number in kilobytes? Or, should we just use `4 * 1024 * 1024 * 1024`?

Comment (sakridge) [reply ledger/src/blockstore_db.rs:848]: uh.. yea. thanks

Comment (sakridge) [reply ledger/src/blockstore_db.rs]: Fixed.

**Comment by sakridge** [comment]
This seems to help, I ran a test with bench-tps overnight, ledger was 480gb and took 10 seconds to open.

**Pull request merged**